src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1948,21 +1948,24 @@ private static void validateStyleAnchors(Set<Node> anchors, String fileName, Str
 
         for (Node anchor : anchors) {
             final String actualUrl;
-            final String expectedUrl;
+            final StringBuilder expectedUrl = new StringBuilder(ruleNumber);
 
             if (position == 1) {
                 actualUrl = XmlUtil.getNameAttributeOfNode(anchor);
-                expectedUrl = ruleNumber;
+
+                if ("google_style.xml".equals(fileName)) {
+                    expectedUrl.insert(0, "a");
```

Comment:
Such strong hack, what is side effects if we apply it to all links ?
Maybe we should apply it to names that are contains only digits. Usually names don't start from digits, and it might be reason why "a" is added upfront, it might make link to work in most browsers